src/unix/linux_like/linux/mod.rs

```diff
@@ -1394,6 +1488,62 @@ cfg_if! {
                 self.sched_period.hash(state);
             }
         }
+
+        impl PartialEq for ptp_sys_offset {
+            fn eq(&self, other: &ptp_sys_offset) -> bool {
+                self.n_samples == other.n_samples &&
+                    self.rsv == other.rsv &&
+                    self.ts[..] == other.ts[..]
+            }
+        }
+        impl Eq for ptp_sys_offset {}
+        impl ::fmt::Debug for ptp_sys_offset {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("ptp_sys_offset")
+                    .field("n_samples", &self.n_samples)
+                    .field("rsv", &self.rsv)
+                    .field("ts", &&self.ts[..])
+                    .finish()
+            }
+        }
+        impl ::hash::Hash for ptp_sys_offset {
+            fn hash<H: ::hash::Hasher>(&self, state: &mut H) {
+                self.n_samples.hash(state);
+                self.rsv.hash(state);
+                self.ts[..].hash(state);
+            }
+        }
```

Comment:
Why not just move this struct from `s_no_extra_traits!` to `s!` so it gets these derived? Same for `ptp_pin_desc`